metadata_layer.py

```diff
@@ -690,13 +698,21 @@ def links(self, arg_links):
                 # An open-access link or open-access rights implies a FormatData object.
                 open_access_link = (link.rel == Hyperlink.OPEN_ACCESS_DOWNLOAD and link.href)
                 # try to deduce if the link is open-access, even if it doesn't explicitly say it is
-                open_access_rights_link = (link.media_type in Representation.BOOK_MEDIA_TYPES 
-                                           and link.href
-                                           and (link.rights_uri or self.default_rights_uri) in RightsStatus.OPEN_ACCESS)
+                rights_uri =  link.rights_uri or self.default_rights_uri
+                open_access_rights_link = (
+                    link.media_type in Representation.BOOK_MEDIA_TYPES 
+                    and link.href
+                    and rights_uri in RightsStatus.OPEN_ACCESS
+                )
                 
                 if open_access_link or open_access_rights_link:
-                    rights_uri = link.rights_uri or self.default_rights_uri
-                    if open_access_link and not rights_uri in RightsStatus.OPEN_ACCESS:
+                    if (open_access_link
+                        and rights_uri != RightsStatus.IN_COPYRIGHT
```

Comment:
This change is tested in the content server but I don't have a test of it in core.